### PR TITLE
updated parser for ESS readout after ICD review

### PR DESCRIPTION
**Pull request by mortenjc**
### Issue reference / description

The branch you merge from should already reference an event-formation-unit github ticket number. You can add a descriptive title, but if an issue is referenced, you don't have to.

## Checklist for submitter

- [ ] Check for conflict with integration test
- [x] Unit tests pass

---

## Nominate for Group Code Review (Anyone can nominate it)
Indicate if you think the code should be reviewed in a Thursday code review session.

- [ ] Recommend for group code review

Also, nominate it on the code_review Slack channel.

### CODE REVIEW

File: src/modules/loki/test/ReadoutGenerator.cpp [hunk 1]
```diff
@@ -23,11 +23,11 @@ uint16_t lokiReadoutDataGen(uint16_t DataSections, uint16_t DataElements, uint8_
   auto DP = (uint8_t *)Buffer;
   //printf("Buffer pointer %p\n", (void *)Buffer);
   auto Header = (ReadoutParser::PacketHeaderV0 *)DP;
-  #ifdef READOUT_EXTRA_PADDING
-  Header->Padding = 0;
-  #endif
-  Header->CookieVersion = 0x00535345;
-  Header->TypeSubType = 0x30;
+    Header->CookieAndType = 0x30535345;
```
Comment (amues) [inline src/modules/loki/test/ReadoutGenerator.cpp:26]: The ownCloud document has 0x52 for `S` (page 3).

Comment (mortenjc) [reply src/modules/loki/test/ReadoutGenerator.cpp:26]: Good thing we're not reviweing that then ;-) 0x53 is the correct value.
Afonso, you're not supposed to be this thorough when reviewing, but then 
again you didn't find the typo in the document name 'readut'  :-)
But thanks for noticing, I will make a correction

File: src/modules/readout/ReadoutParser.cpp [hunk 2]
```diff
@@ -62,7 +71,7 @@ int ReadoutParser::validate(const char *Buffer, uint32_t Size, uint8_t Type) {
     return -ReadoutParser::ESIZE;
   }
 
-  if (Packet.HeaderPtr->TypeSubType != Type) {
+  if ( (Packet.HeaderPtr->CookieAndType >> 24) != Type) {
```
Comment (amues) [inline src/modules/readout/ReadoutParser.cpp:74]: Is the extra space between the two opening parentheses there for a reason?

**Review by amues — APPROVED** [review]

File: src/modules/loki/test/ReadoutGenerator.cpp (hunk 1 above)
Comment (amues) [reply src/modules/loki/test/ReadoutGenerator.cpp:26]: I did miss that 😄

**Comment by mortenjc** [comment]
Forgot to merge

**Pull request merged**